File: src/Java/401-500/458.PoorPigs.java


import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        if (buckets == 1)
            return 0;
        int base = minutesToTest / minutesToDie + 1;
        int r = 1;
        for (int i = 1;i < 100000;i++)
        {
            r *= base;
            if (r >= buckets)
                return i;
        }
        return 0;
    }
}

File: src/Python/401-500/458.PoorPigs.py

class Solution(object):
    def poorPigs(self, buckets, minutesToDie, minutesToTest):
        """
        :type buckets: int
        :type minutesToDie: int
        :type minutesToTest: int
        :rtype: int
        """
        if buckets == 1:
            return 0      
        single = minutesToTest // minutesToDie + 1
        r = 1
        i = 1
        while True:
            r *= single
            if r >= buckets:
                return i
            i += 1
        
if __name__ == '__main__':
    solution = Solution()
    print(solution.poorPigs(1000, 15, 60))  # 5
    print(solution.poorPigs(1000, 12, 60))  # 4
    print(solution.poorPigs(1000, 10, 60))  # 4
    print(solution.poorPigs(1, 1, 1))       # 0
    print(solution.poorPigs(4, 1, 2))       # 2
else:
    pass
